subprojects/docs/src/samples/userguide/java/basic/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,106 @@
+// tag::apply-java-plugin[]
+plugins {
+    java
+}
+
+project.setProperty("sourceCompatibility", "1.8")
+project.setProperty("targetCompatibility", "1.8")
+
+version = "1.2.1"
+// end::apply-java-plugin[]
+
+// tag::java-dependency-mgmt[]
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation("org.hibernate:hibernate-core:3.6.7.Final")
+}
+// end::java-dependency-mgmt[]
+
+// tag::java-basic-test-config[]
+dependencies {
+    testImplementation("junit:junit:4.12")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnit()
+
+    maxHeapSize = "1G"
+}
+// end::java-basic-test-config[]
+
+// tag::practical-integ-test-source-set[]
+sourceSets {
+    create("intTest") {
+        compileClasspath += sourceSets["main"].output
+        runtimeClasspath += sourceSets["main"].output
+    }
+}
+
+val intTestImplementation by configurations.getting {
+    extendsFrom(configurations.implementation)
+}
+
+configurations["intTestRuntimeOnly"].extendsFrom(configurations.runtimeOnly)
+
+dependencies {
+    intTestImplementation("junit:junit:4.12")
+}
+// end::practical-integ-test-source-set[]
+
+// tag::using-custom-doclet[]
+val asciidoclet by configurations.creating
+
+dependencies {
+    asciidoclet("org.asciidoctor:asciidoclet:1.+")
+}
+
+tasks.getByName<Javadoc>("javadoc") {
+    options.docletpath = asciidoclet.files.toList()
+    options.doclet = "org.asciidoctor.Asciidoclet"
+}
+// end::using-custom-doclet[]
+
+
+// tag::skip-tests-condition[]
+tasks["test"].onlyIf { !project.hasProperty("mySkipTests") }
+// end::skip-tests-condition[]
+
+// tag::java-compiler-options[]
+tasks.getByName<JavaCompile>("compileJava") {
+    options.apply {
+        isIncremental = true
+        isFork = true
+        isFailOnError = false
+    }
+}
+// end::java-compiler-options[]
+
+// tag::integ-test-task[]
+val integrationTest by tasks.creating(Test::class) {
+    description = "Runs integration tests."
+    group = "verification"
+
+    testClassesDirs = sourceSets["intTest"].output.classesDirs
+    classpath = sourceSets["intTest"].runtimeClasspath
+    shouldRunAfter("test")
+}
+
+tasks["check"].dependsOn(integrationTest)
+// end::integ-test-task[]
+
+// tag::defining-sources-jar-task[]
+tasks.create<Jar>("sourcesJar") {
+    classifier = "sources"
+    from(sourceSets["main"].allJava)
+}
+// end::defining-sources-jar-task[]
+
+
+// tag::defining-custom-javadoc-task[]
+tasks.create<Javadoc>("testJavadoc") {
```

Comment:
Please use `task("name", Type::class) {}` syntax when creating tasks.

Reply:
@eriwen: @eskatos seemed to suggest `task<Type>(name)` for the _Working with Files_ chapter samples. For example, [here](https://github.com/gradle/gradle/pull/6501/files/ffa334b5c6493fb3372ce50ede92e66c97677f43#r213376492).

Reply:
@pledbrook @eskatos @eriwen in the first PRs, what to use for creating tasks was a bit unclear. To ensure consistency, I'll search for `tasks.create()` and `task(name, SomClass::class)` in the already merged samples and create a PR to consistently use `task<SomeClass>("name")` or `task("name")`.

Reply:
Done: #6719